barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -3817,24 +3866,43 @@ std::vector<Row> AvmTraceBuilder::finalize(uint32_t min_trace_size, bool range_c
 
         dest.incl_mem_tag_err_counts = FF(static_cast<uint32_t>(src.m_tag_err_count_relevant));
 
-        // Calldatacopy/return memory operations are handled differently and are activated by m_sel_op_slice.
+        // TODO: Should be a cleaner way to do this in the future. Perhaps an "into_canoncal" function in
```

Comment:
Candidate for a future refactor, instead of embedding this mem trace "switching" logic in  finalize